### PR TITLE
Add SuperLUMT support 

### LINKED ISSUE #680: Add support for SuperLU MT
https://computation.llnl.gov/sites/default/files/public/cvs_guide.pdf p.269ff

### PR DESCRIPTION
**Pull request by dweindl**
Closes #680.

Only for CMake-based build yet. 

TODO:
- [x] Add wrapper
- [x] Build script for SuperLUMT, SuperLUMT option for sundials, CMake option in AMICI
- [x] Test
- [x] Document
- [x] Allow setting number of threads
- [ ] NewtonSolverSparse::NewtonSolverSparse still uses SUNKLU
- [ ] Enable for Python?
- [ ] Enable on travis?
- [ ] Enable for Matlab?

### CODE REVIEW

File: include/amici/sundials_linsol_wrapper.h [hunk 1]
```diff
@@ -13,7 +14,9 @@
 #include <sunlinsol/sunlinsol_spfgmr.h>
 #include <sunlinsol/sunlinsol_spgmr.h>
 #include <sunlinsol/sunlinsol_sptfqmr.h>
-
+#ifdef SUNDIALS_SUPERLUMT
+#include <sunlinsol/superlumt/fsunlinsol_superlumt.h>
```
Comment (FFroehlich) [inline include/amici/sundials_linsol_wrapper.h]:
```suggestion
#include <sunlinsol/sunlinsol_superlumt.h>
```
I don't think we want to use the fortran headers here

Comment (dweindl) [reply include/amici/sundials_linsol_wrapper.h]: Fair point ;)

**Comment by dweindl** [comment]
Segfaults :(

**Comment by dweindl** [comment]
First quick test with CS_Signalling_ERBB_RAS_AKT (1228 states) not too promising. Fastest with `minDegATPlusA` ordering. Using one thread, Simulation took 2x longer than with KLU. Using 2 threads doubles runtime. Should test with something bigger.

**Comment by dweindl** [comment]
I'd leave it as is for now. Can be used via C++ interface. The other points can be addressed once it's really needed. For me SuperLT_MT does not seem practical.

File: include/amici/defines.h [hunk 2]
```diff
@@ -131,20 +132,13 @@ enum class NonlinearSolverIteration {
...
-};
-
 /** Sensitivity computation mode in steadyStateProblem */
 enum class SteadyStateSensitivityMode {
     newtonOnly,
     simulationFSA
 };
 
-/** State in which the steady state computionat finished */
+/** State in which the steady state computationfinished */
```
Comment (FFroehlich) [inline include/amici/defines.h]:
```suggestion
/** State in which the steady state computation finished */
```

File: include/amici/solver.h [hunk 3]
```diff
@@ -555,17 +555,17 @@ class Solver {
...
+     * @brief sets KLU / SuperLUMT state ordering mode
      * @return
      */
-    StateOrdering getStateOrdering() const;
+    int getStateOrdering() const;
 
     /**
-     * @brief sets KLU state ordering mode (only applies when linsol is set to
-     * amici.AMICI_KLU)
+     * @brief Sets KLU / SuperLUMT state ordering mode (only applies when linsol
```
Comment (FFroehlich) [inline include/amici/solver.h]: should probably specify that KLU and SuperLU have different ordering schemes

**Pull request merged**